client/state/immediate-login/actions.js

```diff
@@ -0,0 +1,23 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+/**
+ * Internal dependencies
+ */
+import { IMMEDIATE_LOGIN_SAVE_STATUS } from 'state/action-types';
+
+/**
+ * Stores immediate link-related information in state so it can be reused later
+ *
+ * @param  {string} reason - Reason for this immediate login
+ * @return {void}
+ */
+export const saveImmediateLoginInformation = reason => dispatch => {
+	dispatch( {
+		type: IMMEDIATE_LOGIN_SAVE_STATUS,
+		reason,
+	} );
+};
```

Comment:
No need for a thunk (async action) here -- you can simply

```jsx
export const saveImmediateLoginInformation = reason => ( {
	type: IMMEDIATE_LOGIN_SAVE_STATUS,
	reason,
} );
```

see https://redux.js.org/basics/actions

Reply:
I'll probably need to do a small followup on this PR soon (to make some minor changes once we start using it) so I'll plan to include the above improvement when I do that also.